openedx/features/philu_courseware/views.py

```diff
@@ -4,80 +4,82 @@
+            'request': request
+        })
+        if serializer.is_valid():
+            serializer.save()
+            return self._get_score_response(chapter_id, status.HTTP_201_CREATED)
 
-    serializer = CompetencyAssessmentRecordSerializer(data=competency_records, context=dict(request=request), many=True)
-    if serializer.is_valid():
-        serializer.save()
-        competency_assessment_score = get_competency_assessments_score(request.user, chapter_id)
-        return Response(
-            {
-                'competency_assessment_score': competency_assessment_score,
-                'message': COMP_ASSESS_RECORD_SUCCESS_MSG
-            }, status=status.HTTP_201_CREATED)
-    else:
-        return Response({'message': serializer.errors}, status=status.HTTP_400_BAD_REQUEST)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
-@api_view(['POST'])
 @view_auth_classes(is_authenticated=True)
-@renderer_classes([JSONRenderer, BrowsableAPIRenderer])
-def revert_user_post_assessment_attempts(request, course_id):
-    user = request.user
-    problem_id = request.data.get('problem_id')
-    validated_problem_id = validate_problem_id(problem_id)
-    reverted_attempts_count = CompetencyAssessmentRecord.objects.revert_user_post_assessment_attempts(
-        problem_id=validated_problem_id, user=user
-    )
-    has_deleted_post_assessment_attempts = reverted_attempts_count > 0
-    if has_deleted_post_assessment_attempts:
-        revert_user_attempts_from_edx(course_id, user, validated_problem_id)
+class RevertPostAssessmentAttemptsAPIView(APIView):
+
+    def post(self, request, course_id):
+        user = request.user
+        problem_id = request.data.get('problem_id')
+        validated_problem_id = validate_problem_id(problem_id)
+        reverted_attempts_count = CompetencyAssessmentRecord.objects.revert_user_post_assessment_attempts(
+            problem_id=validated_problem_id, user=user
+        )
+        has_deleted_post_assessment_attempts = reverted_attempts_count > 0
+        if has_deleted_post_assessment_attempts:
+            self._revert_user_attempts_from_edx(course_id, validated_problem_id)
+
+        return Response(status=status.HTTP_200_OK)
 
-    return Response(status=status.HTTP_200_OK)
+    def _revert_user_attempts_from_edx(self, course_id, problem_usage_key):
+        """
+        :param course_id: str course id
+        :param problem_usage_key: UsageKey problem id
+        """
+        try:
+            user = self.request.user
+            course_id = CourseKey.from_string(course_id)
+            module_state_key = problem_usage_key.map_into_course(course_id)
+            enrollment.reset_student_attempts(
+                course_id=course_id,
+                student=user,
+                module_state_key=module_state_key,
+                requesting_user=user,
+                delete_module=True
+            )
+        except InvalidKeyError:
```

Comment:
The method `reset_student_attempts` also raises value error. Please have a look and tackle it if required.

Reply:
done